alpaca/broker/client.py

```diff
@@ -658,19 +658,15 @@ def download_trade_document_for_account_by_id(
             except HTTPError as http_error:
                 if response.status_code in self._retry_codes:
                     continue
-                if "code" in response.text:
-                    error = response.json()
-                    if "code" in error:
-                        raise APIError(error, http_error)
-                else:
-                    raise http_error
+                raise APIError(http_error) from http_error
 
             # if we got here there were no issues', so response is now a value
             break
 
-        if response is None:
-            # we got here either by error or someone has mis-configured us, so we didn't even try
-            raise Exception("Somehow we never made a request for download!")
+        # we got here either by error or someone has mis-configured us, so we didn't even try
+        assert isinstance(
```

Comment:
this assert also can be replaceable?

Reply:
I'd raise a TypeError here, sounds good?